### ISSUE TITLE
AmazonCognitoYourUserPoolsDemo doesn't sign out properly

### ISSUE THREAD

**Issue by M-Genio** [awslabs/aws-sdk-android-samples#267]
I had a project from less than a year ago that did federated login from facebook, google, and cognito user pools. I referenced it for a new project, but updated my dependencies to the latest `2.7.6`. I found that calling `CognitoUser.currentUser().signOut()` was not actually clearing the cached user.

I noticed in this project the dependency is set to `2.6.8`, which produces the `Didn't find class "org.apache.commons.logging.LogFactory" on path: DexPathList' error.`

Updating the dependency to `2.7.6` resolves the LogFactory issue, but the signOut issue remains. How are we supposed to handle cached user sessions and signing out in the latest version of the `cognitoidentityprovider` sdk

**Comment by minbi** [comment]
Hi @M-Genio ,

The `LogFactory` issue is expected for that version of SDK.

What are you seeing that indicates the cached user is not cleared?
The `signOut()` method clears the memory reference and the stored tokens. [Link for inspection](https://github.com/aws/aws-sdk-android/blob/master/aws-android-sdk-cognitoidentityprovider/src/main/java/com/amazonaws/mobileconnectors/cognitoidentityprovider/CognitoUser.java#L1758)

**Comment by M-Genio** [comment]
@minbi I noticed the 'workaround' for the LogFactory. Thanks!

1) Log in with AuthenticationHandler
2) Call CognitoUserPool.getCurrentUser().signOut()
3) CognitoUserPool.getCurrentUser() still returns the previously signed in user

It used to return null on the userId after signing out, but now it does not. I noticed the getSession() method, which is what probably should be used. I believe it is working as expected.
